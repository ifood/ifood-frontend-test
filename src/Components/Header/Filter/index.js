import React, {useState} from 'react'
import { BsFilterLeft } from 'react-icons/bs'

import {
    Container, SearchContainer, Input, Buttom, Text, Form,
    ParametersContainer, Select, Option, InputParameters
} from './styles'

export default function Filter({search, handleSearchChange, filters}){
    const [show, setShow] = useState(false)
    
    return(
        <Container>
            <SearchContainer>
                <Input placeholder="Buscar" value={search} onChange={event => handleSearchChange(event)}/>
                <Buttom onClick={() => setShow(!show)}>
                    <Text>Filtros</Text>
                    <BsFilterLeft/>
                </Buttom>
            </SearchContainer>
            <Form display={show} onSubmit={() => console.log('ok')}>
                    {filters.map(item =>
                        <ParametersContainer key={item.id}>
                            <Text>{item.name}</Text>
                            {item.values ? <Select> <Option> </Option>
                            {item.values.map(value =>
                                <Option>{value.name}</Option>
                            ) }</Select> : <InputParameters
                                placeholder={item.validation.pattern ? item.validation.pattern : ''}
                                type={item.validation.pattern ? "text" : "number"}
                                pattern={item.validation.pattern ? "[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}" : ""}
                            />}
                        </ParametersContainer>
                    )}
            </Form>
        </Container>
    )
}