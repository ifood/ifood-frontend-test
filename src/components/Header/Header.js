import React from 'react';

const Header = () => ((
  <header>
    <div className="container">
      <h1>Spotifood</h1>
    </div>
  </header>
));

export default Header;
