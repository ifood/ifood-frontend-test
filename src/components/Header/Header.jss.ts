import { makeStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles(
  (theme: Theme) => ({
    title: {
      flexGrow: 1,
      fontSize: '1.5rem',
      fontWeight: theme.typography.fontWeightMedium,
    },
  }),
  { name: 'Header' }
);
