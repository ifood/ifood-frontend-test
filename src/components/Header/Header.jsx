
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const Container = styled.div`
  flex-grow: 1;
`;

const StyledAppBar = styled(AppBar)`
  && {
    color: white;
    background-color: #bc2026;
  }
`;

const Header = (props) => {
  const { title } = props;
  return (
    <Container>
      <StyledAppBar position="static">
        <Toolbar>
          <Typography variant="h4" color="inherit">
            {title}
          </Typography>
        </Toolbar>
      </StyledAppBar>
    </Container>
  );
};

Header.propTypes = {
  title: PropTypes.string,
};

Header.defaultProps = {
  title: '',
};

export default Header;
