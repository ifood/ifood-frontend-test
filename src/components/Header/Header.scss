@import '../../assets/styles/variables';
@import '../../assets/styles/mixins';

.header {
  border-bottom: 1px solid rgba($grey-light, .4);
  padding: .5rem 0;
  width: 100%;
  position: sticky;
  top: 0;
  background-color: white;
  z-index: 2;
  @include lg-max {
    padding: .5rem;
  }
  h1 {
    font-size: 1.4rem;
    font-weight: normal;
    color: $grey;
  }
  .container {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
}
