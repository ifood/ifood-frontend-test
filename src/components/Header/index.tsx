import React, { HTMLAttributes } from "react";

import { Container } from "./styles";
import { ISpotifyUser } from "../../config/interfaces";

interface IHeaderProps extends HTMLAttributes<HTMLDivElement> {
  user: ISpotifyUser | null;
  logOut(): void;
}

const Header: React.FC<IHeaderProps> = ({ user, logOut, ...restProps }) => {
  return (
    <Container>
      <h1>Spotifood</h1>

      { user && (
        <div className="user-info">
          <img src={user.images[0].url} alt={user.display_name} />
          <div>
            <p>{user.display_name}</p>
            <span onClick={logOut}>sair</span>
          </div>
        </div>
      )}
    </Container>
  );
};

export default Header;
