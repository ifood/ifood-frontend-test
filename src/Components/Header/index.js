import React from 'react'

import Filter from './Filter'
import Logo from './Logo'

import { Container } from './styles'

export default function Index({filters, search, handleSearchChange}){
    return(
        <Container>
            <Logo/>
            <Filter filters={filters} search={search} handleSearchChange={handleSearchChange}/>
        </Container>
    )
}