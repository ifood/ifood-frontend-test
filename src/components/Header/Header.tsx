import React from 'react';
import Logo from '../../assets/images/logo.svg';

import './Header.scss';

const Header = () => {
  return (
    <div className="header">
      <div className="container">
        <img src={Logo} alt="Logo"/>
        <h1>Spotifood</h1>
      </div>
    </div>
  )
};

export default Header;
