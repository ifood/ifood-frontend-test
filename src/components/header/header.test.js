import * as React from 'react';
import * as ReactDOM from 'react-dom';
import Header from './header';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

configure({ adapter: new EnzymeAdapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Header />)
});

it('renders Header without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('navBar title text should be "SpotiFood"', () => {
  expect(wrapper.find('#title').text()).toBe('SpotiFood');
});