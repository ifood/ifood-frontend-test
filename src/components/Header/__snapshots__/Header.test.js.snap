// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render() renders the content correctly 1`] = `
<styled.div>
  <Styled(WithStyles(AppBar))
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        color="inherit"
        variant="h4"
      >
        test title
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </Styled(WithStyles(AppBar))>
</styled.div>
`;
